src/fheroes2/gui/ui_dialog.cpp

```diff
@@ -806,6 +809,40 @@ namespace fheroes2
         return false;
     }
 
+    MonsterDialogElement::MonsterDialogElement( const Monster & monster )
+        : _monster( monster )
+    {
+        if ( !_monster.isValid() ) {
+            throw std::logic_error( "Invalid monster provided" );
+        }
+        const Sprite & sprite = AGG::GetICN( ICN::STRIP, 12 );
+        _area = { sprite.width(), sprite.height() };
+    }
+
+    void MonsterDialogElement::draw( Image & output, const Point & offset ) const
+    {
+        Sprite sprite = AGG::GetICN( ICN::STRIP, 12 );
+        renderMonsterFrame( _monster, sprite, { 6, 6 } );
+        Blit( sprite, 0, 0, output, offset.x, offset.y, sprite.width(), sprite.height() );
```

Comment:
An optional suggestion.
We can use `Copy()` here because all monster images don't have any transparent pixels. We can even make the `sprite` to be single-layer:
```suggestion
        Sprite sprite = AGG::GetICN( ICN::STRIP, 12 );
        sprite._disableTransformLayer();
        renderMonsterFrame( _monster, sprite, { 6, 6 } );
        Copy( sprite, 0, 0, output, offset.x, offset.y, sprite.width(), sprite.height() );
```

Reply:
Done